sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/TokenValidatorInternal.cs

```diff
@@ -2,126 +2,77 @@
+            string authorizedPartyKey = tokenSchemaVersion == SupportedTokenSchemaVersions.V2_0 ? ConfigurationManager.AzpKey : ConfigurationManager.AppIdKey;
+            jsonWebToken.TryGetPayloadValue(authorizedPartyKey, out string authorizedPartyValue);
 
-            if (!ConfigurationManager.GetService(token.Payload[(tokenSchemaVersion == SupportedTokenSchemaVersions.V2_0 ? ConfigurationManager.TOKEN_V2_VERIFY : ConfigurationManager.TOKEN_V1_VERIFY)].ToString(), out ServiceInfo serviceInfo))
+            if (configurationManager.ValidateAuthorizationParty(authorizedPartyValue) == false)
             {
                 return (false, null);
             }
 
             try
             {
-                await Task.Run(() =>
-                {
-                    if (MustCacheRefresh())
-                    {
-                        try
-                        {
-                            _cacheKeys.Clear();
-                            _semaphore.Wait();
-
-                            using (HttpClient httpClient = new HttpClient())
-                            {
-                                string openidConfiguration = httpClient.GetStringAsync(new Uri($"{serviceInfo.OpenIdConnectionHost}/common/v2.0/.well-known/openid-configuration", UriKind.Absolute)).Result;
+                string oidcConfigAddress = configurationManager.GetOpenIDConfigurationUrlString(tokenSchemaVersion);
 
-                                AuthenticationEventJsonElement openidConfigurationJson = new AuthenticationEventJsonElement(openidConfiguration);
-                                string jwksUri = openidConfigurationJson.GetPropertyValue("jwks_uri");
-                                string certs = httpClient.GetStringAsync(new Uri(jwksUri, UriKind.Absolute)).Result;
-                                AuthenticationEventJsonElement certsJson = new AuthenticationEventJsonElement(certs);
-                                foreach (AuthenticationEventJsonElement key in certsJson.GetPropertyValue<AuthenticationEventJsonElement>("keys").Elements)
-                                {
-                                    _cacheKeys.Add(new RsaSecurityKey(new RSAParameters { Modulus = FromBase64Url(key.GetPropertyValue("n")), Exponent = FromBase64Url(key.GetPropertyValue("e")) }));
-                                }
-                            }
-                        }
-                        finally
-                        {
-                            _semaphore.Release();
-                        }
-                        _lastSync = DateTime.Now;
-                    }
-                }).ConfigureAwait(false);
+                OpenIdConnectConfiguration oidcConfig =
+                    await OpenIdConnectConfigurationRetriever.GetAsync(
+                        address: oidcConfigAddress,
+                        cancel: CancellationToken.None).ConfigureAwait(false);
```

Comment:
ideally, we should allow passing a CT, so maybe the interface should allow passing it in

Reply:
Let's take it up as a follow up then.